### PR TITLE
The Rabbit Separatist Movement from Easter (BNUYY EDITION)

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Rabbits are a snowflakey mob. They were originally made to be solely for the easter event, with the default state being them holding their cute lil' easter baskets. This PR is created to re-make the rabbit, for a new era. All rabbit code is no longer in easter.dm, and has been rewritten to accommodate it's speshial new widdle code file: rabbits.dm. Everything has been repathed, and rabbit sprites are no longer in easter.dmi, but rather: rabbits.dmi!!! I just had to touch a lot of them to ensure that the rabbit without a basket is standard (as opposed to the rabbit with a basket). A couple of the sprites weren't... correct at all, so those have been updated as well.

I also added rabbits to the Pet Park Holodeck in this PR. It wouldn't make sense for me to have done all these changes without a little bit of appearance in the game.

## Why It's Good For The Game

Rabbits shouldn't just be for easter, and I hope this opens up the door for bunny lovers (like myself) to really expand on this adorable little creacher. This sets the stage for more rabbit content (which I hope to add a lot to as well).

According to all of my testing on local (summoning the easter event, rabbits) the overall functionality appears to be the same. 

## Changelog

:cl:
add: The species of Rabbits signed a declaration saying that they no longer wanted to be a part of Easter.  After a long four-year separatist movement and a lot of bloodshed, they aren't! 
add: While the rabbits, erm, breed like rabbits, you can now pet one of those adorable critters in the holodeck.
fix: If for some reason you wish to kill one of these creatures, the black and brown variants should now have their eyes comically turn into an X (because that wasn't there before).
/:cl:

### CODE REVIEW

**Comment by jughu** [comment]
This is wabbite approved
Easter best holiday

File: code/modules/mob/living/simple_animal/friendly/rabbit.dm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	icon_living = "e_rabbit_white"
+	icon_dead = "e_rabbit_white_dead"
+	icon_prefix = "e_rabbit"
+	speak = list(
+		"Hop into Easter!",
+		"Come get your eggs!",
+		"Prizes for everyone!",
+	)
+	icon_prefix = "e_rabbit"
+	var/initial_egg_amount = 10 ///passed to egg_layer component as how many eggs it starts out as able to lay.
```
Comment (Ryll-Ryll) [inline code/modules/mob/living/simple_animal/friendly/rabbit.dm:51]:
```suggestion
	///passed to egg_layer component as how many eggs it starts out as able to lay.
	var/initial_egg_amount = 10 
```
autodoc goes above the var

**Comment by EricZilla** [comment]
bnuy

File: code/modules/mob/living/simple_animal/friendly/rabbit.dm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	icon_prefix = "s_rabbit"
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+	unsuitable_atmos_damage = 0
+
+/mob/living/simple_animal/rabbit/easter/Initialize(mapload)
+	. = ..()
+	var/eggs_added_from_eating = rand(1, 4)
+	var/max_eggs_held = 8
```
Comment (Ryll-Ryll) [inline code/modules/mob/living/simple_animal/friendly/rabbit.dm:66]: May as well move this to `/mob/living/simple_animal/rabbit/easter`. Also seems a bit odd that they start out with more eggs than their max, but i know that's just how they already were and it doesn't cause any code issues currently. Your call if you wanna change this or not

**Review by Ryll-Ryll — APPROVED** [review]
Tend those 2 tiny reviews and let me know when you're done

**Review by Twaticus — APPROVED** [review]

File: code/modules/mob/living/simple_animal/friendly/rabbit.dm (hunk 2 above)
Comment (san7890) [reply code/modules/mob/living/simple_animal/friendly/rabbit.dm]: Maybe it's so they can lay more eggs (as a surge) when the event starts and then have a lower cap? I'm not interested in rebalancing the easter code in this PR and balancing it, just wanted to move the code around to get rabbits out of easter.dm.

I moved them to the parent.

File: code/modules/mob/living/simple_animal/friendly/rabbit.dm [hunk 3]
```diff
@@ -0,0 +1,78 @@
+//Good ol' rabbits
+
+/mob/living/simple_animal/rabbit
```
Comment (tralezab) [inline code/modules/mob/living/simple_animal/friendly/rabbit.dm:3]: proper dmdoc, short desc of what they do

File: code/modules/mob/living/simple_animal/friendly/rabbit.dm [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	speak = list(
+		"Hop into Easter!",
+		"Come get your eggs!",
+		"Prizes for everyone!",
+	)
+	icon_prefix = "e_rabbit"
+	///passed to egg_layer component as how many eggs it starts out as able to lay.
+	var/initial_egg_amount = 10
+	var/eggs_added_from_eating = rand(1, 4)
+	var/max_eggs_held = 8
```
Comment (tralezab) [inline code/modules/mob/living/simple_animal/friendly/rabbit.dm:54]: dm doc em!

**Review by tralezab — CHANGES_REQUESTED** [review]
dmdoc stuff

File: code/modules/mob/living/simple_animal/friendly/rabbit.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/mob/living/simple_animal/friendly/rabbit.dm:3]: that means the // comment needs to gain one more / and be right above the type- it is missing a general gist

Comment (san7890) [reply code/modules/mob/living/simple_animal/friendly/rabbit.dm]: I looked at what other simple_mobs did for that and put that in the latest commit.

**Pull request merged**